src/core/scope.ml

```diff
@@ -73,11 +73,16 @@ type mode =
   (** Scoping mode for rewriting rule left-hand sides. The constructor carries
       a flag that is set to [true] if {!constructor:Terms.expo.Privat} symbols
       are allowed, and also additional data. *)
-  | M_RHS  of bool * (string, tevar) Hashtbl.t
-  (** Scoping mode for rewriting rule righ-hand sides. The constructor carries
-      a flag that is set to [true] if {!constructor:Terms.expo.Privat} symbols
-      are allowed, and the environment for variables that we known to be bound
-      in the RHS. *)
+  | M_RHS  of
+      { m_r_p             : bool
```

Comment:
Following what is done for M_LHS, introduce a type m_rhs_data with fields prefixed by m_rhs. By the way, you could include the boolean argument of M_LHS in m_lhs_data as well.

Reply:
What's the advantage of doing so? It's a guess but it might be more efficient to avoid creating a new type (and we don't need that type, do we?).

Reply:
I'd even be in favour of inlining the record of `M_LHS`.

Reply:
I don't know what to think of 4be4073, what's your opinion @fblanqui?